src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -355,6 +399,7 @@ private async Task<SearchResult> SearchImpl(SqlSearchOptions sqlSearchOptions, S
                             out string searchParameterHash,
                             out Stream rawResourceStream);
                         numberOfColumnsRead = reader.FieldCount;
+                        avePopulateTime = ((avePopulateTime * readCount) + populateWatch.Elapsed.TotalMilliseconds) / (readCount + 1);
```

Comment:
>avePopulateTime = ((avePopulateTime * readCount) + populateWatch.Elapsed.TotalMilliseconds) / (readCount + 1); [](http://example.com/codeflow?start=24&length=110)

This is too complex for me. I suggest to use .Start() .Stop() to get full duration of operation and calculate avg only when reporting.